ci/test/cargo-test/mzcompose.py

```diff
@@ -7,50 +7,46 @@
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0.
 
-from materialize import ROOT, ci_util, spawn
-from materialize.mzcompose import Composition, Service
+import os
+
+from materialize import spawn
+from materialize.mzcompose import Composition
 from materialize.mzcompose.services import Kafka, Postgres, SchemaRegistry, Zookeeper
 
 SERVICES = [
     Zookeeper(),
-    Kafka(),
+    Kafka(
+        # We need a stable port to advertise, so pick one that is unlikely to
+        # conflict with a Kafka cluster running on the local machine.
+        port="30123:9092",
```

Comment:
why do we need a stable port?

Reply:
Because the port needs to go into the `KAFKA_ADVERTISED_LISTENERS` on the next line, which in turn is necessary because we're running the tests on the host but accessing a Kafka in a container.